api/v1beta1/machinehealthcheck_webhook.go

```diff
@@ -136,15 +136,27 @@ func (m *MachineHealthCheck) validate(old *MachineHealthCheck) error {
 		)
 	}
 
+	allErrs = append(allErrs, m.ValidateCommonFields(specPath)...)
+
+	if len(allErrs) == 0 {
+		return nil
+	}
+	return apierrors.NewInvalid(GroupVersion.WithKind("MachineHealthCheck").GroupKind(), m.Name, allErrs)
+}
+
+// ValidateCommonFields validates UnhealthyConditions NodeStartupTimeout, MaxUnhealthy, and RemediationTemplate of the MHC.
+// It is used to help validate other types which define MachineHealthChecks such as MachineHealthCheckClass and MachineHealthCheckTopology.
+func (m *MachineHealthCheck) ValidateCommonFields(specPath *field.Path) field.ErrorList {
```

Comment:
Should/can we add a disclaimer that this func is only used for our internal validation and is not supposed to be used by external consumers?

Reply:
Added - I guess the alternative is to add this into our /internal package if we want to really keep it out of the API? Given what it does right now I don't think it's a big issue, but that would be the 'right' solution IMO.

Reply:
I think it's fine. Considering we leak the behavior through ValidateCreate and ValidateUpdate anyway and we also kind of don't want to treat them as an API